test/unit/repositories/test_catalogue_category.py

```diff
@@ -424,11 +432,19 @@ def test_delete_with_children_catalogue_items(test_helpers, database_mock, catal
     """
     catalogue_category_id = str(ObjectId())
 
-    # Mock count_documents to return 0 (children catalogue categories not found)
-    test_helpers.mock_count_documents(database_mock.catalogue_categories, 0)
-    # Mock count_documents to return 1 (children catalogue items found)
-    test_helpers.mock_count_documents(database_mock.catalogue_items, 1)
-
+    # Mock `find_one` to return no child catalogue category document
+    test_helpers.mock_find_one(database_mock.catalogue_categories, None)
+    # pylint: disable=duplicate-code
+    # Mock `find_one` to return the child catalogue item document
+    test_helpers.mock_find_one(
+        database_mock.catalogue_items,
+        {
+            "_id": CustomObjectId(str(ObjectId())),
+            "catalogue_category_id": CustomObjectId(catalogue_category_id),
+            **FULL_CATALOGUE_ITEM_A_INFO,
```

Comment:
```suggestion
             **FULL_CATALOGUE_ITEM_A_INFO,
            "_id": CustomObjectId(str(ObjectId())),
            "catalogue_category_id": CustomObjectId(catalogue_category_id),
```
The `catalogue_category_id` is being overwritten here.